assignments/mini-libc/tests/test_io.c

```diff
@@ -0,0 +1,451 @@
+/* SPDX-License-Identifier: BSD-3-Clause */
+
+#include <stdlib.h>
+#include <unistd.h>
+#include <fcntl.h>
+#include <sys/stat.h>
+#include <errno.h>
+
+#include "./graded_test.h"
+
+#define RDONLY_FILE "./file_RDONLY"
+#define NON_EXISTENT_FILE "./file_NONEXISTENT"
+#define RW_FILE "./file_RW"
+#define CREATE_FILE "./file_CREATE"
+#define DIR_FILE "./dir"
+#define EMPTY_FILE "./file_EMPTY"
+
+static int test_open_non_existent_file(void)
+{
+	int fd;
+
+	fd = open(NON_EXISTENT_FILE, O_RDONLY);
+
+	return fd == -1 && errno == ENOENT;
+}
+
+static int test_open_invalid_access_mode(void)
+{
+	int fd;
+
+	fd = open(RDONLY_FILE, O_WRONLY);
+
+	return fd == -1 && errno == EACCES;
+}
+
+static int test_open_file_as_directory(void)
+{
+	int fd;
+
+	fd = open(RDONLY_FILE, O_DIRECTORY | O_RDONLY);
+
+	return fd == -1 && errno == ENOTDIR;
+}
+
+static int test_open_directory_for_writing(void)
+{
+	int fd;
+
+
+	fd = open(DIR_FILE, O_WRONLY);
+	return fd == -1 && errno == EISDIR;
+}
+
+static int test_open_force_invalid_creation(void)
+{
+	int fd;
+
+	fd = open(RDONLY_FILE, O_CREAT | O_EXCL);
+
+	return fd == -1 && errno == EEXIST;
+}
+
+static int test_open_close_existent_file(void)
+{
+	int fd;
+
+	fd = open(RDONLY_FILE, O_RDONLY);
+	if (fd < 0)
+		return 0;
+
+	fd = close(fd);
+	if (fd < 0)
+		return 0;
+
+	return 1;
+}
+
+static int test_open_close_create_file(void)
+{
+	int fd;
+
+	fd = open(CREATE_FILE, O_CREAT | O_RDWR, S_IRUSR);
```

Comment:
Shouldn't this be created with the mode `S_IRUSR | S_IWUSR`?
Since the file is opened with `O_RDWR`, it gives a `permission denied` error if it's created just with reading permissions.

Reply:
You're right, sorry, the created file was left from a previous run and that's why it failed on a read-write open.